common/src/java/org/apache/hadoop/hive/conf/HiveConf.java

```diff
@@ -597,6 +597,8 @@ private static void populateLlapDaemonVarsSet(Set<String> llapDaemonVarsSetLocal
     HADOOPNUMREDUCERS("mapreduce.job.reduces", -1, "", true),
 
     // Metastore stuff. Be sure to update HiveConf.metaVars when you add something here!
+    METASTOREDBTYPE("hive.metastore.db.type", "DERBY", new StringSet("DERBY", "ORACLE", "MYSQL", "MSSQL", "POSTGRES"),
+        "Type of database used for the metastore"),
```

Comment:
can we determine this using DatabaseProduct.determineDatabaseProduct instead of a config ?
If we keep config for now, can you also mention that this is needed for the functionality in this jira ?